site/source/docs/porting/simd.rst

```diff
@@ -35,6 +35,45 @@ When porting native SIMD code, it should be noted that because of portability co
 
 SIMD-related bug reports are tracked in the `Emscripten bug tracker with the label SIMD <https://github.com/emscripten-core/emscripten/issues?q=is%3Aopen+is%3Aissue+label%3ASIMD>`_.
 
+===========================
+Optimization considerations
```

Comment:
swizzle semantics match arm's TBL, on x86 it's a couple more instructions to have the same out of bounds behavior. Use a swizzle with a constant, engines can pattern match that to elide the add on x86.

Reply:
Added, let me know what you think.